src/plugin/fieldViews/RepeaterFieldView.ts

```diff
@@ -0,0 +1,65 @@
+import type { Node } from "prosemirror-model";
+import type { Decoration, DecorationSet } from "prosemirror-view";
+import type { FieldDescriptions } from "../types/Element";
+import type { BaseFieldDescription } from "./FieldView";
+import { FieldType } from "./FieldView";
+
+export const repeaterFieldName = "repeater" as const;
+
+export const createRepeaterField = <FDesc extends FieldDescriptions<string>>(
+  fields: FDesc
+) => ({
+  type: repeaterFieldName,
+  fields,
+});
+
+export interface RepeaterFieldDescription<
+  FDesc extends FieldDescriptions<string>
+> extends BaseFieldDescription<unknown> {
+  type: typeof repeaterFieldName;
+  fields: FDesc;
+}
+
+/**
+ * A FieldView representing a node that contains user-defined child nodes.
+ *
+ * Offers methods to add, remove, and move nodes.
```

Comment:
Should this be 'add, remove, and destroy'? (Matching the method names). What is the distinction between remove and destroy here?

Reply:
Great question. These were added as I tried to imagine what this FieldView would be responsible for – but I've left them unimplemented as there will be no meaningful way to test them until they're wired up to Prosemirror's state.

I've removed them for now, as no interface requires them yet, and they're probably more confusing that enlightening!